packages/@ama-sdk/client-angular/src/api-angular-client.ts

```diff
@@ -66,7 +69,17 @@ export class ApiAngularClient implements ApiClient {
   }
 
   /** @inheritdoc */
-  public extractQueryParams<T extends { [key: string]: any }>(data: T, names: (keyof T)[]): { [p in keyof T]: string; } {
+  public extractQueryParams<T extends { [key: string]: any }>(data: T, names: (keyof T)[]): { [p in keyof T]: string; };
+  /** @inheritdoc */
+  public extractQueryParams<T extends { [key: string]: any }>(data: T, names: { [K in keyof T]?: Omit<ParamSerialization, 'value'> }): { [p in keyof T]?: ParamSerialization; };
+  /** @inheritdoc */
+  public extractQueryParams<T extends { [key: string]: any }>(
+    data: T,
+    names: (keyof T)[] | { [K in keyof T]?: Omit<ParamSerialization, 'value'> }
+  ): { [p in keyof T]: string; } | { [p in keyof T]?: ParamSerialization; } {
+    if (Array.isArray(names)) {
+      return extractQueryParams(data, names);
+    }
     return extractQueryParams(data, names);
```

Comment:
Why is it different to 

```suggestion
    return extractQueryParams(data, names);
```
?

Reply:
But it should not change the return type of `extractQueryParams`.
Did you face type issue without the `Array.isArray`?
(I was actually triggered because adding JS process to handle TypeScript type issue is generally not the good practice :S)